pkg/notifiers/error.go

```diff
@@ -0,0 +1,45 @@
+package notifiers
+
+// NotifierError represents an error that occurs in notifier calls.
+type NotifierError struct {
+	// msg provides additional information or context about the error.
+	msg string
+
+	// err holds the underlying error.
+	err error
+}
+
+// NewNotifierError creates and returns a new instance of NotifierError.
+//
+// Parameters:
+//   - msg: A string providing additional information or context about the error.
+//     This message will be visible by users, and it should not expose any
+//     sensitive information.
+//   - err: The underlying error that caused the extraction failure.
+//
+// Returns:
+//   - An initialized pointer to an NotifierError struct with the given parameters.
+func NewNotifierError(msg string, err error) *NotifierError {
+	return &NotifierError{
+		msg: msg,
+		err: err,
+	}
+}
+
+var _ error = (*NotifierError)(nil)
+
+func (e *NotifierError) Error() string {
```

Comment:
How about adding a test for this. I know it's simple but it will be good to see how if behaves and how it differs from "normal" error. Also do we need `String()` method? Should we handle `%+v` option like errors does and then they print a stack trace? In our case we could print the message + err

Reply:
I have removed the implementation of the `error` interface. I do not see a reason for using it.